### PR TITLE
I did the navigation, internal screen and request

### PR DESCRIPTION
**Pull request by IagoWag**
# 🚨 Fill in the fields below

## Description
```
~> the implementation of the navigation of the stacknavigation type was carried out,
the internal screen and the api request were also performed.
```

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

```

```

## Checklist

- [x] This is an example checked

- [x] PRODUCT
  - [x] Put PR link on ticket
  - [x] Move the ticket
  - [x] Put the time spent on this task on ticket
- [x] DEVELOPERS
  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
 

- [x] REVIEW
  - [x] Add reviewers: The entire team and most influential people in the area/project
  - [x] Assignees myself to it
 
🚨 - `Approve: Wait at least one influential in the project or more experienced` 
 
# FRONTEND

## Design

[
marvel_shot
](https://cdn.discordapp.com/attachments/855096737438236694/873255270262902784/marvel_shot.png)

## Screenshot or video of your application

🚨 - `Fill with your application`

https://user-images.githubusercontent.com/102339646/178124197-ab41843b-7358-4aa2-ad02-137462fbefcc.mp4

### CODE REVIEW

**Review by JeanSilvany — APPROVED** [review]
Good job!

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+import Routes from './navigation';
 
-// import { Container } from './styles';
+function App() {
+    return (
+        <NavigationContainer>
+            <Routes />
+        </NavigationContainer>
+    );
+}
```
Comment (IbraDajani) [inline src/App.tsx:12]: Switch it to an arrow function and dont forget to export by default

File: src/navigation/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import Home from '../screens/Home';
+
+const {Navigator, Screen} = createStackNavigator();
+
+const animatedNavigation: StackNavigationOptions = {
+    gestureDirection: 'horizontal',
+    gestureEnabled: true,
+    headerShown: false,
+    cardStyleInterpolator: CardStyleInterpolators.forHorizontalIOS,
+};
```
Comment (IbraDajani) [inline src/navigation/index.tsx:17]: Move this to a controller file like `useNavigationController.ts`

File: src/screens/Home/index.tsx [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 import Separator from '../../components/Separator';
 import Text from '../../components/Text';
 import Row from '../../components/Row';
-import RowBetween from '../../components/RowBetween';
 import {Container, HeaderImage, OptionButton} from './styles';
 import CharacterCard from '../../components/CharacterCard';
-import {mockedData as item} from './mock';
+import {useNavigation} from '@react-navigation/native';
+import {ResponseItemProps, ResponseProps} from './types';
+import axios from 'axios';
```
Comment (IbraDajani) [inline src/screens/Home/index.tsx:18]: We are using the request from the `api.ts` service file... remove this axios import

File: src/screens/Home/index.tsx [hunk 4]
```diff
@@ -27,7 +35,29 @@ const filterOptions = [
...
+            console.log(error);
+        } finally {
+            setLoading(false);
+        }
+    }, []);
+
+    useEffect(() => {
+        handleFechtData();
+    }, [handleFechtData]);
+
```
Comment (IbraDajani) [inline src/screens/Home/index.tsx:60]: Comment on each section to organize the code by its use type

File: src/screens/Internal/index.tsx [hunk 5]
```diff
@@ -1,10 +1,72 @@
...
+                                <Text size={20}>Back</Text>
+                            </Row>
+                        </Pressable>
+                    }
+                />
+            </CharacterImage>
+
+            <CharacterInfoConfig>
+                <CharacterHeaderInfo>
+                    <CharacterName isBold size={30}>
```
Comment (IbraDajani) [inline src/screens/Internal/index.tsx:42]: move those text props to an `attrs({ })` prop in the `styles.ts`

**Review by IbraDajani — CHANGES_REQUESTED** [review]

File: src/screens/Internal/index.tsx (hunk 5 above)
Comment (IbraDajani) [reply src/screens/Internal/index.tsx:42]: move every styled component with props usage to the `attrs` prop in the styles file

File: src/App.tsx (hunk 1 above)
Comment (IagoWag) [reply src/App.tsx]: resolved in commit a71461893b6fea906309d9e3507e0c49e0d139f5

File: src/navigation/index.tsx (hunk 2 above)
Comment (IagoWag) [reply src/navigation/index.tsx]: resolved in commit https://github.com/IagoWag/rickandmorty/commit/a71461893b6fea906309d9e3507e0c49e0d139f5

File: src/screens/Home/index.tsx (hunk 3 above)
Comment (IagoWag) [reply src/screens/Home/index.tsx]: resolved in commit https://github.com/IagoWag/rickandmorty/commit/a71461893b6fea906309d9e3507e0c49e0d139f5

File: src/screens/Home/index.tsx (hunk 4 above)
Comment (IagoWag) [reply src/screens/Home/index.tsx]: resolved in commit https://github.com/IagoWag/rickandmorty/commit/a71461893b6fea906309d9e3507e0c49e0d139f5

File: src/screens/Internal/index.tsx (hunk 5 above)
Comment (IagoWag) [reply src/screens/Internal/index.tsx]: resolved in commit https://github.com/IagoWag/rickandmorty/commit/a71461893b6fea906309d9e3507e0c49e0d139f5

**Review by IbraDajani — APPROVED** [review]

**Comment by IbraDajani** [comment]
LGTM!!!

**Pull request merged**